sycl/plugins/native_cpu/pi_native_cpu.cpp

```diff
@@ -0,0 +1,1308 @@
+    return ReturnValue(PI_DEVICE_MEM_CACHE_TYPE_READ_WRITE_CACHE);
+  case PI_DEVICE_INFO_GLOBAL_MEM_CACHELINE_SIZE:
+    // TODO : CHECK
+    return ReturnValue(pi_uint32{64});
+  case PI_DEVICE_INFO_GLOBAL_MEM_CACHE_SIZE:
+    // TODO : CHECK
+    return ReturnValue(pi_uint64{0});
+  case PI_DEVICE_INFO_GLOBAL_MEM_SIZE:
+    // TODO : CHECK
+    return ReturnValue(pi_uint64{0});
+  case PI_DEVICE_INFO_MAX_CONSTANT_BUFFER_SIZE:
+    // TODO : CHECK
+    return ReturnValue(pi_uint64{0});
+  case PI_DEVICE_INFO_MAX_CONSTANT_ARGS:
+    // TODO : CHECK
+    return ReturnValue(pi_uint32{64});
+  case PI_DEVICE_INFO_LOCAL_MEM_TYPE:
+    // TODO : CHECK
+    return ReturnValue(PI_DEVICE_LOCAL_MEM_TYPE_LOCAL);
+  case PI_DEVICE_INFO_ERROR_CORRECTION_SUPPORT:
+    return ReturnValue(pi_bool{false});
+  case PI_DEVICE_INFO_PROFILING_TIMER_RESOLUTION:
+    // TODO : CHECK
+    return ReturnValue(size_t{0});
+  case PI_DEVICE_INFO_BUILT_IN_KERNELS:
+    // TODO : CHECK
+    return ReturnValue("");
+  case PI_DEVICE_INFO_PRINTF_BUFFER_SIZE:
+    // TODO : CHECK
+    return ReturnValue(size_t{1024});
+  case PI_DEVICE_INFO_PREFERRED_INTEROP_USER_SYNC:
+    return ReturnValue(pi_bool{false});
+  case PI_DEVICE_INFO_PARTITION_AFFINITY_DOMAIN:
+    return ReturnValue(pi_device_affinity_domain{0});
+  case PI_DEVICE_INFO_MAX_MEM_ALLOC_SIZE:
+    // TODO : CHECK
+    return ReturnValue(pi_uint64{0});
+  case PI_DEVICE_INFO_EXECUTION_CAPABILITIES:
+    // TODO : CHECK
+    return ReturnValue(
+        pi_device_exec_capabilities{PI_DEVICE_EXEC_CAPABILITIES_KERNEL});
+  case PI_DEVICE_INFO_PROFILE:
+    return ReturnValue("FULL_PROFILE");
+  case PI_DEVICE_INFO_REFERENCE_COUNT:
+    // TODO : CHECK
+    return ReturnValue(pi_uint32{0});
+  case PI_DEVICE_INFO_BUILD_ON_SUBDEVICE:
+    return ReturnValue(pi_bool{0});
+  case PI_DEVICE_INFO_ATOMIC_64:
+    return ReturnValue(pi_bool{0});
+  case PI_EXT_ONEAPI_DEVICE_INFO_BFLOAT16_MATH_FUNCTIONS:
+    return ReturnValue(pi_bool{0});
+
+    CASE_PI_UNSUPPORTED(PI_DEVICE_INFO_MAX_NUM_SUB_GROUPS)
+    CASE_PI_UNSUPPORTED(PI_DEVICE_INFO_SUB_GROUP_INDEPENDENT_FORWARD_PROGRESS)
+    CASE_PI_UNSUPPORTED(PI_DEVICE_INFO_SUB_GROUP_SIZES_INTEL)
+    CASE_PI_UNSUPPORTED(PI_DEVICE_INFO_IL_VERSION)
+
+    // Intel-specific extensions
+    CASE_PI_UNSUPPORTED(PI_DEVICE_INFO_PCI_ADDRESS)
+    CASE_PI_UNSUPPORTED(PI_DEVICE_INFO_GPU_EU_COUNT)
+    CASE_PI_UNSUPPORTED(PI_DEVICE_INFO_GPU_SLICES)
+    CASE_PI_UNSUPPORTED(PI_DEVICE_INFO_GPU_SUBSLICES_PER_SLICE)
+    CASE_PI_UNSUPPORTED(PI_DEVICE_INFO_GPU_EU_COUNT_PER_SUBSLICE)
+    CASE_PI_UNSUPPORTED(PI_DEVICE_INFO_MAX_MEM_BANDWIDTH)
+    CASE_PI_UNSUPPORTED(PI_DEVICE_INFO_IMAGE_SRGB)
+    CASE_PI_UNSUPPORTED(PI_EXT_ONEAPI_DEVICE_INFO_MAX_GLOBAL_WORK_GROUPS)
+    CASE_PI_UNSUPPORTED(PI_EXT_ONEAPI_DEVICE_INFO_MAX_WORK_GROUPS_1D)
+    CASE_PI_UNSUPPORTED(PI_EXT_ONEAPI_DEVICE_INFO_MAX_WORK_GROUPS_2D)
+    CASE_PI_UNSUPPORTED(PI_EXT_ONEAPI_DEVICE_INFO_MAX_WORK_GROUPS_3D)
```

Comment:
Not a blocking issue, but it would be a good idea to check if any of the `CASE_PI_UNSUPPORTED` cases are used to check aspects. The reason is we use `sycl-ls --verbose` to check aspects in our e2e testing, so having this throw an exception for any aspect check may break testing and should be prioritized.

Chances are you already run e2e testing locally though and have considered this, in which case you can just ignore (and resolve) this comment.

Reply:
The same should likely be applied to various `get_info` queries. We currently don't use them all in `sycl-ls` but there are requests to do so.

Reply:
I've updated the plug in so that no exceptions are thrown when `sycl-ls --verbose` is run